app/(articles)/[displayStyle]/[page]/page.tsx

```diff
@@ -0,0 +1,62 @@
+import { getLastArticlePage, getLastBlogPage } from '../../../../api-helpers/articles';
+import { closeConnection, openConnection } from '../../../../api-helpers/prisma/db';
+import { BlogsGrid } from '../../../../components/BlogsGrid/BlogsGrid';
+import { BlogsList } from '../../../../components/BlogsList/BlogsList';
+import { ButtonAsLink } from '../../../../components/ButtonAsLink/ButtonAsLink';
+import { DisplayStyleSwitch } from '../../../../components/DisplayStyleSwitch/DisplayStyleSwitch';
+import { MAX_PAGES } from '../../../../constants';
+import { getPagesArray } from '../../../../utils/array-utils';
+
+import Styles from './page.module.scss';
+
+import type { DisplayStyle } from '../../../../types';
+
+type HomePageProps = {
+  readonly params: {
+    readonly displayStyle: DisplayStyle;
+    readonly page: string;
+  };
+};
+
+export default function HomePage({ params }: HomePageProps) {
+  const { displayStyle, page } = params;
+
+  return (
+    <>
+      <h2 className={Styles.heading}>Wszystkie artykuły</h2>
+
+      <div className={Styles.buttons}>
+        <ButtonAsLink href="/zglos-serwis" icon="icon-plus">
+          Dodaj serwis
+        </ButtonAsLink>
+
+        <DisplayStyleSwitch value={displayStyle} />
+      </div>
```

Comment:
To nie powinien być layout.tsx? Wtedy unikasz tego `ts-expect-error` poniżej.
W praktyce nigdy nie miałem tego problemu z async components bo nigdy ich nie używam w środku innych komponentów – tylko przez layout.